src/Components/Blueprints/BlueprintVersionFilter.tsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+
+import {
+  Dropdown,
+  DropdownItem,
+  DropdownList,
+  MenuToggle,
+  MenuToggleElement,
+} from '@patternfly/react-core';
+
+interface blueprintVersionFilterProps {
+  blueprintVersionFilter?: string | undefined;
+  setBlueprintVersionFilter?: React.Dispatch<
+    React.SetStateAction<string | undefined>
+  >;
+}
+
+export const ALL_VERSIONS_CONST = 'All Versions';
+export const NEWEST_VERSION_CONST = 'Newest';
+const LATEST_VERSION = -1;
+
+export const getBlueprintVersionFilterAPIValue = (
```

Comment:
This seems a bit over-engineered to me. I think that using Redux (and taking more advantage of Typescript types) will help simplify this a bit. 

Once we have a selector, we’ll be able to do something like
```Typescript
const blueprintVersionFilter = selectBlueprintVersionFilter(...)
```

I would have the type in the store be something like
```Typescript
type BlueprintVersionFilter = ‘all’ | ‘newest’
```

Then for this function you could just use a switch case, something like
```Typescript
const getBlueprintVersionFilter = (blueprintVersionFilter: BlueprintVersionFilter) => {
  switch (blueprintVersionFilter) {
    case 'all':
      return undefined;
    case 'newest':
      return '-1';
    default:
      return undefined;
  }
};
```

Reply:
Yeah I've completely rewritten it for it all be stored in the slice.